server/db/index.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const util = require('util');
+const path = require('path');
+const fs = require('fs');
+
+const readFile = util.promisify(fs.readFile);
+const writeFile = util.promisify(fs.writeFile);
+
+const moviesPath = path.resolve('server/db/movies.json');
+
+// Reads and returns contents of movies.json
+async function readMovies() {
+  return readFile(moviesPath)
+    .then((contents) => {
+      return JSON.parse(contents);
+    });
+}
+
+// Replaces the contents of movies.json
+async function writeMovies(movies) {
+  const json = JSON.stringify(movies, null, 2);
+  return writeFile(moviesPath, json);
+}
+
+// Gets a list of each unique movie
+async function getAllTitles() {
+  return readMovies()
+    .then((movies) => {
+      const names = [];
+      movies.forEach((movie) => {
+        if (!names.includes(movie.name)) {
+          names.push(movie.name);
+        }
+      });
+      return names;
+    });
+}
+
+// Gets a filtered list of all movies that meet all search conditions
+async function searchMovies(query) {
+  return readMovies()
+    .then((movies) => {
+      const matches = [];
+      movies.forEach((movie) => {
+        // If movie name in query even partially exists in any movie name
+        let isMovieMatch = true;
+        if (movie.name && query.name) {
+          isMovieMatch = movie.name.toLowerCase()
+            .includes(query.name.toLowerCase());
+        }
```

Comment:
An other way to write this would be

```js
const isMovieMatch = (!movie.name || !query.name)
  && movie.name.toLowerCase().includes(query.name.toLowerCase());
```

Don't feel like you have to do this though, just whatever you find easier to follow.